### DISCUSSION TITLE
Task Grouping: How to use `*` and `**` wildcards?

### DISCUSSION THREAD

**Discussion by powerman** [jdx/mise#2907] [Q&A]
I can't see any difference between `*` and `**` wildcards. Is there any working example for this?

Doc says:

> For example running `mise run test:**:local` will match `test:units:local`, `test:integration:local` and `test:e2e:happy:local`.

But `mise run 'test:*:local'` will match the same tasks.

**Comment by Marukome0743** [comment]
This has now been fixed on `main` by [PR #11581](https://github.com/jdx/mise/pull/11581).

Task-name wildcards now treat `:` as a group boundary:

- `test:*:local` matches one intermediate group, such as `test:unit:local`.
- It no longer matches deeper tasks such as `test:e2e:happy:local`.
- `test:**:local` recursively matches zero or more intermediate groups, including both `test:local` and deeper descendants.

The same matching rules are used for direct task execution, dependencies, monorepo tasks, workspace tasks, and task-information commands. Other glob syntax such as braces, `?`, and character classes also stays within a single task group.

Existing configurations that intentionally relied on `*` matching recursively should change those patterns to `**`.

This change was merged after v2026.8.0, so it should be included in the next release.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
